build.rs

```diff
@@ -0,0 +1,102 @@
+use serde::{Deserialize, Serialize};
+use std::{fs, io::Write};
+use toml::de;
+
+/// Reads and deserializes test data from tests/test.toml
+/// and generates rust integration test included in
+/// tests/generated_tests.rs during the build process
+fn main() {
+    let out_dir = std::env::var("OUT_DIR").unwrap();
+    let destination = std::path::Path::new(&out_dir).join("test.rs");
+    let mut f = std::fs::File::create(&destination).unwrap();
+
+    let test_list = read_tests();
+
+    for test in test_list.tests {
+        let test_name = test.name
+            .trim()
+            .replace(" ", "_")
+            .replace(&[',', '.'][..], "");
+
+        write!(
+            f,
+            "
+#[test]
```

Comment:
Can you debug this? I could be totally off base with the right way to do things in Rust, but it seems like it would be better to just read the values from tests.toml, but create a separate function that can be compiled and have debug points set. Or maybe this somehow is debuggable at runtime in vscode thanks to unwrap?

Reply:
I've separated out the boiler plate logic of the test into in a common file/crate in tests (each file in the tests folder is compiled into it's own crate). Additionally, the generated file is now part of the project. This should allow better debugging.